src/main/java/pl/javastart/task/UniversityApp.java

```diff
@@ -27,10 +62,34 @@ public void createLecturer(int id, String degree, String firstName, String lastN
      * @param name       - nazwa przedmiotu (np. "Podstawy programowania")
      * @param lecturerId - identyfikator prowadzącego. Musi zostać wcześniej utworzony za pomocą metody {@link #createLecturer(int, String, String, String)}
      */
+
     public void createGroup(String code, String name, int lecturerId) {
+        if (groupSize >= groups.length) {
+            groups = Arrays.copyOf(groups, groups.length * 2);
+        }
+
+        Group group = findGroup(code);
+        Lecturer lecturer = findLecturer(lecturerId);
+
+        if (lecturer == null) {
+            System.out.printf("Prowadzący o id %d nie istnieje\n", lecturerId);
+        } else if (group == null) {
+            groups[groupSize] = new Group(code, name, lecturerId);
+            groupSize++;
+        } else {
+            System.out.printf("Grupa %s już istnieje\n", code);
+        }
 
     }
 
+    private Group findGroup(String code) {
+        for (int i = 0; i < groupSize; i++) {
+            if (groups[i].getCode() == code) {
```

Comment:
Pamiętaj o porównywaniu Stringów przez metodę equals(), a nie ==
to chyba jeszcze przed Tobą w kursie, ale możesz już teraz zerknąć na lekcję: https://app.javastart.pl/kurs/java/java-podstawy-temat-22-klasa-string/lekcja/java-podstawy-string-stringbuilder-i-stringbuffer